app/components/works/form_component.rb

```diff
@@ -29,5 +31,17 @@ def work
     def work_version
       work_form.model.fetch(:work_version)
     end
+
+    def doi_field
+      return "https://doi.org/#{Doi.for(druid)}." if doi
+
+      # create DOI URL or placeholder for works in collections that allow DOI assignment
+      collection = work.collection
+      return unless (collection.doi_option == 'depositor-selects' && work.assign_doi) || collection.doi_option == 'yes'
```

Comment:
It looks like this duplicates some logic in `work.assign_doi?` Can you extract that into a separate method (`will_assign_doi?` perhaps) that we can use here?

Reply:
I moved this logic into a new method as you suggested and put it in `draft_work_form.rb`. Does that make sense?